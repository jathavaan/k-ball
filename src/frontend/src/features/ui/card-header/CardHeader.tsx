import { StyledCardHeader } from "./cardHeader.style.ts";
import { CardHeaderProps } from "./cardHeader.types.ts";

export const CardHeader = (props: CardHeaderProps) => {
  return (
    <StyledCardHeader {...props} variant="h4">
      {props.headerText}
    </StyledCardHeader>
  );
};
