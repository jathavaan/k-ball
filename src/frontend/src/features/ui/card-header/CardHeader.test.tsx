import { render, screen } from "@testing-library/react";
import { CardHeader } from "./CardHeader";

describe("Tests for CardHeader", () => {
  it("Should render the header text", () => {
    const mockProps = {
      headerText: "Test Header",
    };

    render(<CardHeader {...mockProps} />);

    expect(screen.getByText(mockProps.headerText)).toBeDefined();
  });

  it("Should match CardHeader snapshot", () => {
    const mockProps = {
      headerText: "Test Header",
    };

    const { asFragment } = render(<CardHeader {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
