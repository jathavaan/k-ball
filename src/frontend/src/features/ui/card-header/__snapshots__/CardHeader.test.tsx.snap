// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for CardHeader > Should match CardHeader snapshot 1`] = `
<DocumentFragment>
  <h4
    class="MuiTypography-root MuiTypography-h4 css-1gp0puz-MuiTypography-root"
  >
    Test Header
  </h4>
</DocumentFragment>
`;
